framework/src/araui/core/ui_core.c

```diff
@@ -211,51 +212,57 @@ static ui_error_t _ui_process_widget_recur(ui_widget_body_t *widget, uint32_t dt
 		return UI_INVALID_PARAM;
 	}
 
-	if (widget->tween_cb) {
-		if (widget->tween_info.t > widget->tween_info.d) {
-			widget->tween_info.t = widget->tween_info.d;
-			widget->tween_cb((ui_widget_t)widget, widget->tween_info.t);
-			widget->tween_cb = UI_NULL;
+	ui_widget_stack_init();
+	ui_widget_stack_push(widget);
 
-			if (ui_request_callback(_ui_call_tween_finished_cb, widget) != UI_OK) {
-				UI_LOGE("Error: cannot make a request to call.tween_finished_cb!\n");
+	while (!ui_widget_is_stack_empty()) {
+		curr_widget = ui_widget_stack_pop();
+		if (curr_widget->tween_cb) {
```

Comment:
You'd better check `curr_widget` is NULL or not for avoiding dereferencing.

Reply:
Thanks, I add checking procedure of `curr_widget`.